src/core/effects/qgsgloweffect.cpp

```diff
@@ -78,9 +78,15 @@ void QgsGlowEffect::draw( QgsRenderContext &context )
   dtProps.ramp = ramp;
   QgsImageOperation::distanceTransform( im, dtProps );
 
-  if ( mBlurLevel > 0 )
+  double blurLevel = context.convertToPainterUnits( mBlurLevel, mBlurUnit, mBlurMapUnitScale );
+  if ( blurLevel <= 16.0 )
   {
-    QgsImageOperation::stackBlur( im, mBlurLevel );
+    QgsImageOperation::stackBlur( im, blurLevel );
+  }
+  else
+  {
+    QImage *imb = QgsImageOperation::gaussianBlur( im, blurLevel );
```

Comment:
This leaks (But actually, QgsImageOperation::gaussianBlur shouldn't be returning a pointer anyway. It could return a value and avoid all the memory management handling since QImage is implicitly shared)